### ISSUE TITLE
windowrulev2 option to close a window

### ISSUE THREAD

**Issue by Kommynct** [hyprwm/Hyprland#5014]
### Description

I have certain windows that appear that in sway, I have set to automatically close

for_window [class="annoyingthing.exe" window_type="dialog" title="Idle Time-Out"] kill 

^ that makes the idle notifier that pops up close instantly because all it's telling me is to push ok to close it.

What would be the hyprland equivalent of this command? If there isn't one, i'd like that functionality

**Comment by vaxerski** [comment]
you can script this no?

**Comment by Kommynct** [comment]
the problem I have with coming up with a script at all is that hyprctl dispatch closewindow annoyingthing.exe would close the whole app, not just the idle timeout popup, but if you have a suggestion i'd love to see it, i'm just stuck on this during my switch from sway

**Comment by vaxerski** [comment]
oh, so it doesn't actually close it but just "hides" the window? That's confusing, why is it named `kill`?

**Comment by Kommynct** [comment]
No, it does close it, i just can't find a way to distinguish the whole .exe and the popup using closewindow

I need both the class and the title to be matched at least to achieve that effect, and there seems to be no way to match title for closewindow

Furthermore, I want it to happen right when it pops up, without polling, and I don't know of an event based way to do that, i'm new to hyprland obviously, sorry

**Comment by paul-gra** [comment]
I think, what you could do is to create a special workspace, where you move these kinds of windows with a `windowrulev2` and `movetoworkspacesilent`.

**Comment by Kommynct** [comment]
I don't see how that helps, again I'm trying to close the window but not the whole app

**Comment by paul-gra** [comment]
My bad, I did not read correctly. I did not realize that you need to close those windows or interact with those.
For now, I would take vaxerski's suggestion and script this behaviour with a custom script, as described in [the corresponding wiki entry](https://wiki.hyprland.org/configuring/expanding-functionality/).

**Comment by paul-gra** [comment]
My bad, I did not read correctly and did not realize you need to interact with those windows.
So using `closewindow` with the **adress** of the window you want to close kill the parent window, too? If yes, did you try to use 'pass' dispatcher to pass enter or whatever to the window of interest?

**Comment by Kommynct** [comment]
I want to kill a window that happens to share a class with another window, but not a title, as soon as it appears, without killing the entire class, and without affecting other classes.

I don't think the pass dispatcher has anything to do with that, and am left mostly confused by your comment.

**Comment by paul-gra** [comment]
Okay, to further elaborate:
I thought you already read [this wiki entry](https://wiki.hyprland.org/configuring/expanding-functionality/), that describes how you can you a custom script involving `socat`, as scripts and dispatchers were already mentioned.
`/tmp/hypr/$HYPRLAND_INSTANCE_SIGNATURE/.socket2.sock` prints a new line `openwindow>><window_address>,<workspace>,<window_class>,<window_title>`  every time a window is opened. You can use these bits of information together with `hyprctl dispatch closewindow` to selectively kill a window with given title **and** class by parsing and specifying the address of the window to kill rather than title or class. 
As an alternative to `hyprland dispatch closewindow`, `hyprland dispatch pass` may be viable, as the window of interest may interpret a keypress such as Enter or Escape as 'Okay".
Both ways, you need a custom script that listens to hyprland events with `socat` or similar.
`hyprland dispatch closewindow` can **not** only take the window class as argument, as described in the respective [wiki entry](https://wiki.hyprland.org/configuring/dispatchers/).

**Comment by Kommynct** [comment]
Thank you, that makes a lot of sense, and I totally missed that wiki entry

That is certainly a good workaround, but I do believe this should be a non-scripted function built into the compositor, so, i'll leave the issue open, i'll post my script here once i've made it so that anybody else searching for this has an easy solution!

**Comment by vaxerski** [comment]
Hello there!

This issue has been closed, as we are moving from issues to discussions. More about this can be found [here](https://wiki.hyprland.org/Contributing-and-Debugging/Issue-Guidelines/) and in #9854.

Firstly, please make sure this issue is still relevant in latest Hyprland. If so, we ask you to open a [discussion](https://github.com/hyprwm/Hyprland/discussions) (please read the discussion guidelines first, in the pinned post)

Thank you for your understanding!
